Core/include/Acts/TrackFitting/Chi2Fitter.hpp

```diff
@@ -352,13 +349,28 @@ class Chi2Fitter {
 
         // add a full TrackState entry multi trajectory
         // (this allocates storage for all components, we will set them later)
-        result.lastTrackIndex = result.fittedStates->addTrackState(
-            ~(TrackStatePropMask::Smoothed | TrackStatePropMask::Filtered),
-            result.lastTrackIndex);
+
+        size_t currentTrackIndex = SIZE_MAX;
+        if (updateNumber == 0) {
+          result.lastTrackIndex =
+              result.fittedStates
+                  ->addTrackState(  // which lastTrackIndex is this?
+                      ~(TrackStatePropMask::Smoothed |
+                        TrackStatePropMask::Filtered),
+                      result.lastTrackIndex);
+          currentTrackIndex = result.lastTrackIndex;
+        } else {
+          result.fittedStates->visitBackwards(
+              result.lastTrackIndex, [&](auto proxy) {
+                if (&proxy.referenceSurface() == surface) {
+                  currentTrackIndex = proxy.index();
+                }
+              });
+        }
 
         // now get track state proxy back
         auto trackStateProxy =
-            result.fittedStates->getTrackState(result.lastTrackIndex);
+            result.fittedStates->getTrackState(currentTrackIndex);
```

Comment:
Currently, if we do not create a trackstate in (`if` branch) and also do not find a trackstate (`else` branch), this should fail, because `currentTrackState` is still `SIZE_MAX`?
Or do I overlook something?

Reply:
yes, I wanted to see if we ever manage to miss I surface in the first run. I now added a `VERBOSE` to get notified, when an additional surface is set. Probably better than a crash.